src/main/java/org/radarbase/appserver/controller/FcmDataMessageController.java

```diff
@@ -233,5 +233,21 @@ public ResponseEntity deleteDataMessagesForUser(
         this.dataMessageService.removeDataMessagesForUser(projectId, subjectId);
         return ResponseEntity.ok().build();
     }
+
+    @PreAuthorize(
+            AuthConstantsUtil.PERMISSION_ON_SUBJECT_MEASUREMENT_CREATE
+                    + AuthConstantsUtil.ACCESSOR
+                    + AuthConstantsUtil.PROJECT_ID
+                    + ", "
+                    + AuthConstantsUtil.ACCESSOR
+                    + AuthConstantsUtil.SUBJECT_ID
+                    + ")")
+    @DeleteMapping("/" + PathsUtil.MESSAGING_DATA_PATH + "/{id}")
```

Comment:
please add project id and subject id in the path as they will be used for authorization.